packages/legend-graph/src/helpers/MappingResolutionHelper.ts

```diff
@@ -155,7 +155,9 @@ export const getLeafSetImplementations = (
     return undefined;
   }
   if (setImp instanceof OperationSetImplementation) {
-    return setImp.leafSetImplementations;
+    return mapping.classMappings.filter(
```

Comment:
See my comment on the expected behavior.
I think we can just remove this if statement and return the setImpl. 
We may need to rename this function tho because it is not really fetching the leafes of setImpl.